### PR TITLE
ignored:list

### PR DESCRIPTION
**Pull request by mshanemc**
### What does this PR do?
the last of the source commands from toolbelt

### What issues does this PR fix or reference?
@W-10449080@

### CODE REVIEW

File: test/nuts/ignored_list.nut.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+ */
+import * as fs from 'fs';
+import * as os from 'os';
+import * as path from 'path';
+import { execCmd, TestSession } from '@salesforce/cli-plugins-testkit';
+import { expect } from 'chai';
+import { AuthStrategy } from '@salesforce/cli-plugins-testkit/lib/hubAuth';
+import { SourceIgnoredResults } from '../../src/commands/force/source/ignored/list';
+
+describe('force:source:ignored:list', () => {
```
Comment (WillieRuemmele) [inline test/nuts/ignored_list.nut.ts:15]: should we test the "no ignored files" possibility?

File: messages/ignored_list.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+  "description": "check your local project package directories for forceignored files",
+  "examples": ["$ sfdx force:source:ignored", "$ sfdx force:source:ignored --sourcepath force-app"],
+  "flags": {
+    "sourcepath": "file or directory of files that the command checks for foreceignored files"
+  },
+  "invalidSourcePath": "File or directory '%s' doesn't exist in your project. Specify one that exists and rerun the command."
```
Comment (WillieRuemmele) [inline messages/ignored_list.json:7]: is this message used?

**Review by shetzel — APPROVED** [review]

File: messages/ignored_list.json (hunk 2 above)
Comment (mshanemc) [reply messages/ignored_list.json:7]: good call...not anymore.  SDR is gonna handle that

**Comment by WillieRuemmele** [comment]
QA:
with https://github.com/forcedotcom/source-deploy-retrieve/pull/553 linked on a stock `dreamhouse-lwc` project

❌  `force:source:ignored:list` previously included all of the `__tests__` directories, now it doesn't show them
✅ : no results found
✅ : JSON structure the same, but missing files from first point above
✅ : handles nested `.forceignores` as long as `-p` points to the parent directory of the nested `.forecignore` (toolbelt didn't recognize any new `.forceignores`)
❓ : didn't read nested `.forceignore` it discovered (same as toolbelt)

**Comment by WillieRuemmele** [comment]
QA: 
✅ : includes `__tests__` correctly
✅ : JSON
✅ : no results
✅ : `-p` behavior
✅ : `-p <file doesn't exist>` errors correctly

**Pull request merged**